plugins/modules/katello_content_view.py

```diff
@@ -95,106 +98,113 @@
-                cvc = component.copy()
-                cvc['content_view'] = find_content_view(module, name=component['content_view'], organization=entity_dict['organization'])
-                cvc_matched = None
-                for _cvc in current_cvcs:
-                    if _cvc.content_view.id == cvc['content_view'].id:
-                        cvc_matched = _cvc
-                force_update = list()
-                if 'version' in component:
-                    cvc['version'] = find_content_view_version(module, cvc['content_view'], version=component['version'])
-                    cvc['latest'] = False
-                    if cvc_matched and cvc_matched.latest:
-                        # When changing to latest=False & version is the latest we must send 'content_view_version' to the server
-                        force_update.append('content_view_version')
-                if cvc_matched:
-                    cvc['composite_content_view'] = content_view_entity
-                    cvc_dict = sanitize_entity_dict(cvc, cvc_map)
-                    cvc_changed = naildown_entity_state(ContentViewComponent, cvc_dict, cvc_matched, 'present', module, force_update=force_update)
-                    current_cvcs.remove(cvc_matched)
-                    if cvc_changed:
-                        changed = cvc_changed
-                else:
-                    for attr in ['latest', 'version']:
-                        if attr not in cvc:
-                            cvc[attr] = None
-                    ContentViewComponent(composite_content_view=content_view_entity, content_view=cvc['content_view'],
-                                         latest=cvc['latest'], content_view_version=cvc['version']).add()
-                    changed = True
-        for cvc in current_cvcs:
+    # Also only act for composite content views
+    if (module.state == 'present' or (module.state == 'present_with_defaults' and changed)) and content_view_entity['composite'] and components is not None:
+        if not changed:
+            content_view_entity['content_view_components'] = entity['content_view_components']
+        current_cvcs = content_view_entity.get('content_view_components', [])
+
+        components_to_add = []
+        ccv_scope = {'composite_content_view_id': content_view_entity['id']}
+        for component in components:
+            cvc = {
+                'content_view': module.find_resource_by_name('content_views', name=component['content_view'], params=scope),
+                'latest': component['latest'],
+            }
+            cvc_matched = next((item for item in current_cvcs if item['content_view']['id'] == cvc['content_view']['id']), None)
+            if not cvc['latest']:
+                search = "content_view_id={},version={}".format(cvc['content_view']['id'], component['content_view_version'])
+                cvc['content_view_version'] = module.find_resource('content_view_versions', search=search, thin=True)
```

Comment:
`find_resource` inside a for loop makes me a bit concerned. Hopefully we've already ruled out using only a single API call?

Reply:
At least not so easy.